docs/source/1.0/spec/core/http-traits.rst

```diff
@@ -932,6 +934,67 @@ many HTTP client and server implementations enforce limits in practice.
 Carefully consider the maximum allowed length of each member that is bound to
 an HTTP query string or path.
 
+.. _httpQueryParams-trait:
+
+``httpQueryParams`` trait
+=========================
+
+Summary
+    Binds a map of key-value pairs to be added as query string parameters.
+Trait selector
+    .. code-block:: none
+
+        structure > member
+        :test(> map > member[id|member=value] > :test(string, collection > member > string))
+
+    The ``httpQueryParams`` trait can be applied to ``structure`` members
+    that target a ``map`` of ``string``, or a ``map`` of ``list``/``set`` of
+    ``string``.
+
+Value type
+    Annotation trait.
+Conflicts with
+   :ref:`httpLabel-trait`, :ref:`httpHeader-trait`, :ref:`httpQuery-trait`,
+   :ref:`httpPrefixHeaders-trait`, :ref:`httpPayload-trait`,
+   :ref:`httpResponseCode-trait`
+
+The following example defines an operation that optionally sends the
+target input map as query string parameters in an HTTP request:
+
+.. tabs::
+
+    .. code-tab:: smithy
+
+        @readonly
+        @http(method: "GET", uri: "/things")
+        operation ListThings {
+            input: ListThingsInput,
+            output: ListThingsOutput, // omitted for brevity
+        }
+
+        structure ListThingsInput {
+            @httpQueryParams()
+            myParams: MapOfStrings,
+        }
+
+        map MapOfStrings {
+            key: String,
+            value: String
+        }
+
+.. rubric:: ``httpQueryParams`` is only used on input
+
+``httpQueryParams`` is ignored when resolving the HTTP bindings of an operation's
+output or an error. This means that if a structure that contains members
+marked with the ``httpQueryParams`` trait is used as the top-level output structure
+of an operation, then those members are sent as part of the
+:ref:`protocol-specific document <http-protocol-document-payloads>` sent in
+the body of the response.
+
+.. rubric:: Serialization rules
+
+See the :ref:`httpQuery-trait` serialization rules that define how the keys and values of the
```

Comment:
Do we need more info here specific to maps? Like are map keys escaped at all if they contain "&" or "="? Are key-value pairs treated exactly like as if they were explicitly bound using httpQuery?